### PR TITLE
feat(azure): added support for multiple azure MSI profiles

### PR DESCRIPTION
**Pull request by pacificcode**
Added support for multiple Azure MSI profiles. When selecting Azure as the profile auth-provider the user is prompted to enter the clientID for the MSI (managed identity) they would like to use for this profile. The clientID is stored in the profile config file i.e. .dsv.yml.
the clientID is retrieved from the config when ever azure authentication is required.

### CODE REVIEW

**Review by sheldonhull — CHANGES_REQUESTED** [review]
Nice! Don't forget to include changie entry that would be written for the customer. That's what triggers any future release.  And also update base branch from main (button on PR or run sync) cause must be up to date from main to merge.

File: commands/cli-config.go [hunk 1]
```diff
@@ -684,16 +684,25 @@ func handleCliConfigInitCmd(vcli vaultcli.CLI, args []string) int {
...
 	authType := viper.GetString(cst.AuthType)
+	clientID := ""
 	if authType == "" {
 		authType, err = promptAuthType()
 		if err != nil {
 			vcli.Out().WriteResponse(nil, errors.New(err))
 			return 1
 		}
 		viper.Set(cst.AuthType, authType)
+		if authType == string(auth.FederatedAzure) {
```
Comment (andrii-zakurenyi) [inline commands/cli-config.go]: Please add another case into the switch below and move prompting for client ID to that case.
This `if`-block is only for getting authentication type if it was not provided via flags.

Comment (pacificcode) [reply commands/cli-config.go]: good catch, thanks. This modification is done

**Review by sheldonhull — APPROVED** [review]
Nice work!

**Pull request merged**